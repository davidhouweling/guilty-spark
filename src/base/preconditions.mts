export const Preconditions = {
  checkExists<T>(value: T, errorMessage?: string): NonNullable<T> {
    if (value == null) {
      throw new Error(errorMessage ?? "Value cannot be null");
    }

    return value;
  },
};
