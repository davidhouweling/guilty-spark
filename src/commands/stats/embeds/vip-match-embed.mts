import type { GameVariantCategory } from "halo-infinite-api";
import type { PlayerStats } from "./base-match-embed.mjs";
import { BaseMatchEmbed } from "./base-match-embed.mjs";

export class VIPMatchEmbed extends BaseMatchEmbed<GameVariantCategory.MultiplayerVIP> {
  override getPlayerObjectiveStats(stats: PlayerStats<GameVariantCategory.MultiplayerVIP>): Map<string, string> {
    return new Map([
      ["VIP kills", stats.VipStats.VipKills.toString()],
      ["VIP Assists", stats.VipStats.VipAssists.toString()],
      ["Kills as VIP", stats.VipStats.KillsAsVip.toString()],
      ["Times selected as VIP", stats.VipStats.TimesSelectedAsVip.toString()],
      ["Max killing spree as VIP", stats.VipStats.MaxKillingSpreeAsVip.toString()],
      ["Longest Time as VIP", this.haloService.getReadableDuration(stats.VipStats.LongestTimeAsVip)],
      ["Time as VIP", this.haloService.getReadableDuration(stats.VipStats.TimeAsVip)],
    ]);
  }
}
