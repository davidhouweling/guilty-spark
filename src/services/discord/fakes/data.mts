import type {
  APIApplicationCommandInteraction,
  APIMessage,
  APIMessageComponentButtonInteraction,
  APIModalSubmitInteraction,
  APIPingInteraction,
  RESTPostAPIChannelMessagesThreadsResult,
} from "discord-api-types/v10";
import {
  ButtonStyle,
  ComponentType,
  MessageFlags,
  ApplicationCommandOptionType,
  ChannelType,
  GuildMemberFlags,
  InteractionType,
  Locale,
  MessageType,
} from "discord-api-types/v10";
import type { QueueData } from "../discord.mjs";

export const apiMessage: APIMessage = {
  type: MessageType.Default,
  content: "Hello, world!",
  mentions: [],
  mention_roles: [],
  attachments: [],
  embeds: [],
  timestamp: "2024-12-06T12:03:09.182000+00:00",
  edited_timestamp: null,
  components: [],
  id: "1314562775950954626",
  channel_id: "1299532381308325949",
  author: {
    id: "000000000000000001",
    username: "soundmanD",
    avatar: "e803b2f163fda5aeba2cf4820e3a6535",
    discriminator: "0850",
    global_name: null,
  },
  mention_everyone: false,
  pinned: false,
  tts: false,
};

export const pingInteraction: APIPingInteraction = {
  id: "fake-id",
  type: InteractionType.Ping,
  application_id: "fake-application-id",
  token: "fake-token",
  version: 1,
  app_permissions: "",
  authorizing_integration_owners: {},
  entitlements: [],
};

const fakeBaseInteraction = {
  app_permissions: "fake-permissions",
  application_id: "fake-application-id",
  authorizing_integration_owners: {},
  context: 0,
  entitlements: [],
  id: "fake-id",
  locale: Locale.EnglishUS,
  member: {
    deaf: false,
    joined_at: "2024-05-11T11:45:17.722000+00:00",
    mute: false,
    permissions: "fake-permissions",
    roles: [],
    user: {
      avatar: null,
      discriminator: "0",
      global_name: null,
      id: "fake-user-id",
      username: "soundmanD",
    },
    flags: GuildMemberFlags.CompletedOnboarding,
  },
  token: "fake-token",
};

const fakeBaseAPIApplicationCommandInteraction: Omit<APIApplicationCommandInteraction, "type" | "data"> = {
  ...fakeBaseInteraction,
  channel: {
    guild_id: "fake-channel-guild-id",
    id: "fake-channel-id",
    type: ChannelType.GuildText,
  },
  channel_id: "fake-channel-id",
  version: 1,
};

export const applicationCommandInteractionStatsNeatQueue: APIApplicationCommandInteraction = {
  ...fakeBaseAPIApplicationCommandInteraction,
  type: InteractionType.ApplicationCommand,
  data: {
    id: "1296081783443685377",
    name: "stats",
    options: [
      {
        name: "neatqueue",
        options: [
          {
            name: "channel",
            type: 7,
            value: "1299532381308325949",
          },
          {
            name: "queue",
            type: 4,
            value: 1418,
          },
        ],
        type: 1,
      },
    ],
    resolved: {
      channels: {
        "1299532381308325949": {
          id: "1299532381308325949",
          name: "ü•âresults",
          permissions: "2230813650837056",
          type: 0,
        },
      },
    },
    type: 1,
  },
};

export const applicationCommandInteractionStatsMatch: APIApplicationCommandInteraction = {
  ...fakeBaseAPIApplicationCommandInteraction,
  type: InteractionType.ApplicationCommand,
  data: {
    id: "1300004385459408960",
    name: "stats",
    options: [
      {
        name: "match",
        options: [
          {
            name: "id",
            type: ApplicationCommandOptionType.String,
            value: "d81554d7-ddfe-44da-a6cb-000000000ctf",
          },
        ],
        type: 1,
      },
    ],
    type: 1,
  },
};

export const buttonClickInteraction: APIMessageComponentButtonInteraction = {
  ...fakeBaseInteraction,
  channel: {
    guild_id: "fake-channel-guild-id",
    id: "fake-channel-id",
    last_message_id: "fake-last-message-id",
    name: "general",
    nsfw: false,
    parent_id: "fake-parent-id",
    position: 0,
    rate_limit_per_user: 0,
    topic: null,
    type: ChannelType.GuildText,
  },
  channel_id: "fake-channel-id",
  data: { component_type: ComponentType.Button, custom_id: "btn_yes" },
  message: {
    application_id: "fake-application-id",
    attachments: [],
    author: {
      avatar: null,
      avatar_decoration_data: null,
      bot: true,
      discriminator: "2015",
      global_name: null,
      id: "fake-guilty-spark-id",
      public_flags: 524288,
      username: "Guilty Spark",
    },
    channel_id: "fake-channel-id",
    components: [
      {
        components: [
          {
            custom_id: "btn_yes",
            label: "Yes",
            style: ButtonStyle.Primary,
            type: ComponentType.Button,
          },
          {
            custom_id: "btn_no",
            label: "No",
            style: ButtonStyle.Danger,
            type: ComponentType.Button,
          },
        ],
        type: ComponentType.ActionRow,
      },
    ],
    content: "",
    edited_timestamp: null,
    embeds: [
      {
        description: "Some description",
        fields: [
          {
            inline: false,
            name: "Field title",
            value: "Field value",
          },
        ],
        title: "Embed title",
      },
    ],
    flags: MessageFlags.Ephemeral,
    id: "fake-message-id",
    interaction: {
      id: "fake-interaction-id",
      name: "fake-interaction-name",
      type: InteractionType.ApplicationCommand,
      user: {
        avatar: null,
        avatar_decoration_data: null,
        discriminator: "0",
        global_name: "fake-user-global-name",
        id: "fake-user-id",
        username: "fake-username",
      },
    },
    interaction_metadata: {
      authorizing_integration_owners: {},
      id: "fake-interaction-metadata-id",
      type: InteractionType.ApplicationCommand,
      user: {
        avatar: null,
        avatar_decoration_data: null,
        discriminator: "0",
        global_name: "fake-user-global-name",
        id: "fake-user-id",
        username: "fake-username",
      },
    },
    mention_everyone: false,
    mention_roles: [],
    mentions: [],
    pinned: false,
    position: 0,
    timestamp: "2025-01-10T09:42:25.495000+00:00",
    tts: false,
    type: 20,
    webhook_id: "fake-webhook-id",
  },
  type: InteractionType.MessageComponent,
  version: 1,
};

export const modalSubmitInteraction: APIModalSubmitInteraction = {
  ...fakeBaseInteraction,
  channel: {
    guild_id: "fake-channel-guild-id",
    id: "fake-channel-id",
    last_message_id: "fake-last-message-id",
    name: "general",
    nsfw: false,
    parent_id: "fake-parent-id",
    position: 0,
    rate_limit_per_user: 0,
    topic: null,
    type: ChannelType.GuildText,
  },
  channel_id: "fake-channel-id",
  data: {
    components: [
      {
        components: [{ custom_id: "text_input", type: ComponentType.TextInput, value: "Hello!" }],
        type: 1,
      },
    ],
    custom_id: "text_input_modal",
  },
  message: {
    application_id: "fake-application-id",
    attachments: [],
    author: {
      avatar: null,
      avatar_decoration_data: null,
      bot: true,
      discriminator: "2015",
      global_name: null,
      id: "fake-guilty-spark-id",
      public_flags: 524288,
      username: "Guilty Spark",
    },
    channel_id: "fake-channel-id",
    components: [
      {
        components: [
          {
            custom_id: "btn_modal",
            label: "Modal",
            style: ButtonStyle.Secondary,
            type: ComponentType.Button,
          },
        ],
        type: ComponentType.ActionRow,
      },
    ],
    content: "",
    edited_timestamp: null,
    embeds: [],
    flags: 64,
    id: "fake-message-id",
    interaction: {
      id: "fake-interaction-id",
      name: "fake-interaction-name",
      type: InteractionType.ApplicationCommand,
      user: {
        avatar: null,
        avatar_decoration_data: null,
        discriminator: "0",
        global_name: "fake-user-global-name",
        id: "fake-user-id",
        username: "fake-username",
      },
    },
    interaction_metadata: {
      authorizing_integration_owners: {},
      id: "fake-interaction-metadata-id",
      type: InteractionType.ApplicationCommand,
      user: {
        avatar: null,
        avatar_decoration_data: null,
        discriminator: "0",
        global_name: "fake-user-global-name",
        id: "fake-user-id",
        username: "fake-username",
      },
    },
    mention_everyone: false,
    mention_roles: [],
    mentions: [],
    pinned: false,
    position: 0,
    timestamp: "2025-01-24T09:41:38.252000+00:00",
    tts: false,
    type: 20,
    webhook_id: "fake-webhook-id",
  },
  type: InteractionType.ModalSubmit,
  version: 1,
};

function aFakeNeatQueueMessageWith(opts: Partial<APIMessage> = {}): APIMessage {
  return {
    type: MessageType.Default,
    content: "",
    mentions: [],
    mention_roles: [],
    attachments: [],
    embeds: opts.embeds ?? [
      {
        title: "üèÜ Winner For Queue#1 üèÜ",
        color: 16711680,
        timestamp: "2024-12-06T12:03:08.786000+00:00",
        fields: [
          {
            name: "Eagle",
            value:
              "<@000000000000000001> *-30.3* **(993.2)**\n<@000000000000000002> *-30.1* **(978.5)**\n<@000000000000000003> *-30.0* **(966.1)**\n<@000000000000000004> *-29.6* **(926.6)**",
            inline: true,
          },
          {
            name: "__Cobra__",
            value:
              "<@000000000000000005> *+28.1* **(1221.0)**\n<@000000000000000006> *+29.5* **(1081.8)**\n<@000000000000000007> *+29.7* **(1063.2)**\n<@000000000000000008> *+32.8* **(755.4)**",
            inline: true,
          },
        ],
      },
    ],
    timestamp: "2024-12-06T12:03:09.182000+00:00",
    edited_timestamp: null,
    components: [],
    id: "1314562775950954626",
    channel_id: "1299532381308325949",
    author: {
      id: "857633321064595466",
      username: "NeatQueue",
      avatar: "e803b2f163fda5aeba2cf4820e3a6535",
      discriminator: "0850",
      public_flags: 65536,
      flags: 65536,
      bot: true,
      banner: null,
      accent_color: null,
      global_name: null,
      avatar_decoration_data: null,
    },
    pinned: false,
    mention_everyone: false,
    tts: false,
    ...opts,
  };
}

export const channelMessages: APIMessage[] = [
  aFakeNeatQueueMessageWith({
    embeds: [
      {
        title: "üèÜ Winner For Queue#10 üèÜ",
        color: 16711680,
        timestamp: "2024-12-05T11:52:20.193000+00:00",
        fields: [
          {
            name: "__Eagle__",
            value:
              "<@000000000000000001> *+29.8* **(1235.8)**\n<@000000000000000002> *+31.6* **(1052.1)**\n<@000000000000000003> *+32.5* **(983.2)**\n<@000000000000000004> *+32.6* **(956.2)**",
            inline: true,
          },
          {
            name: "Cobra",
            value:
              "<@000000000000000005> *-32.5* **(1117.3)**\n<@000000000000000006> *-32.2* **(1099.1)**\n<@000000000000000007> *-31.5* **(1036.6)**\n<@000000000000000008> *-31.3* **(1012.9)**",
            inline: true,
          },
        ],
      },
    ],
    timestamp: "2024-12-05T11:52:21.235000+00:00",
    id: "1314197670398525493",
  }),
  aFakeNeatQueueMessageWith({
    embeds: [
      {
        title: "üèÜ Winner For Queue#7 üèÜ",
        color: 16711680,
        timestamp: "2024-12-06T11:05:39.576000+00:00",
        fields: [
          {
            name: "Eagle",
            value:
              "<@000000000000000009> *-28.2* **(1033.4)**\n<@000000000000000008> *-28.1* **(1023.5)**\n<@000000000000000007> *-27.9* **(1008.7)**\n<@000000000000000006> *-27.8* **(996.1)**",
            inline: true,
          },
          {
            name: "__Cobra__",
            value:
              "<@000000000000000010> *+26.4* **(1289.1)**\n<@000000000000000011> *+27.0* **(1262.8)**\n<@000000000000000012> *+29.1* **(1054.5)**\n<@000000000000000013> *+29.5* **(1052.3)**",
            inline: true,
          },
        ],
      },
    ],
    timestamp: "2024-12-06T11:05:41.120000+00:00",
    id: "1314548313730846792",
  }),
  aFakeNeatQueueMessageWith({
    embeds: [
      {
        title: "üèÜ Winner For Queue#4 üèÜ",
        color: 16711680,
        timestamp: "2024-12-06T09:55:16.814000+00:00",
        fields: [
          {
            name: "Eagle",
            value:
              "<@000000000000000010> *-29.6* **(1057.0)**\n<@000000000000000008> *-29.3* **(1033.2)**\n<@000000000000000005> *-29.2* **(1022.9)**\n<@000000000000000001> *-28.9* **(992.6)**",
            inline: true,
          },
          {
            name: "__Cobra__",
            value:
              "<@000000000000000012> *+27.8* **(1262.7)**\n<@000000000000000014> *+28.4* **(1203.9)**\n<@000000000000000016> *+29.2* **(1128.3)**\n<@000000000000000018> *+31.7* **(880.2)**",
            inline: true,
          },
        ],
      },
    ],
    timestamp: "2024-12-06T09:55:17.673000+00:00",
    id: "1314530599309934632",
  }),
  aFakeNeatQueueMessageWith({
    embeds: [
      {
        title: "üèÜ Winner For Queue#3 üèÜ",
        color: 16711680,
        timestamp: "2024-12-06T08:24:07.415000+00:00",
        fields: [
          {
            name: "__Eagle__",
            value:
              "<@000000000000000001> *+29.2* **(1234.9)**\n<@000000000000000003> *+29.8* **(1175.5)**\n<@000000000000000005> *+31.0* **(1062.6)**\n<@000000000000000007> *+32.1* **(951.3)**",
            inline: true,
          },
          {
            name: "Cobra",
            value:
              "<@000000000000000002> *-32.2* **(1231.7)**\n<@000000000000000004> *-31.0* **(1114.8)**\n<@000000000000000006> *-30.7* **(1086.6)**\n<@000000000000000008> *-28.3* **(848.5)**",
            inline: true,
          },
        ],
      },
    ],
    timestamp: "2024-12-06T08:24:08.804000+00:00",
    id: "1314507661211074611",
  }),
  aFakeNeatQueueMessageWith(),
  aFakeNeatQueueMessageWith({
    embeds: [
      {
        color: 16711680,
        timestamp: "2024-12-06T06:24:07.415000+00:00",
        fields: [
          {
            name: "Some fake value",
            value: "just doing this to for tests",
            inline: true,
          },
        ],
      },
    ],
  }),
  {
    type: MessageType.Default,
    content: "",
    mentions: [],
    mention_roles: [],
    attachments: [],
    embeds: [],
    timestamp: "2024-12-06T12:03:09.182000+00:00",
    edited_timestamp: null,
    components: [],
    id: "1314562775950954626",
    channel_id: "1299532381308325949",
    author: {
      id: "000000000000000001",
      username: "soundmanD",
      avatar: "e803b2f163fda5aeba2cf4820e3a6535",
      discriminator: "0850",
      public_flags: 65536,
      flags: 65536,
      global_name: null,
    },
    pinned: false,
    mention_everyone: false,
    tts: false,
  },
];

export const discordNeatQueueData: QueueData = {
  message: {
    type: 0,
    content: "",
    mentions: [],
    mention_roles: [],
    attachments: [],
    embeds: [
      {
        title: "üèÜ Winner For Queue#777 üèÜ",
        color: 16711680,
        timestamp: "2024-11-26T11:30:00.000000+00:00",
        fields: [
          {
            name: "Eagle",
            value:
              "<@000000000000000001> *-31.5* **(1131.7)**\n" +
              "<@000000000000000002> *-31.4* **(1117.9)**\n" +
              "<@000000000000000003> *-30.4* **(1017.7)**\n" +
              "<@000000000000000004> *-29.8* **(966.3)**",
            inline: true,
          },
          {
            name: "__Cobra__",
            value:
              "<@000000000000000005> *+29.6* **(1195.0)**\n" +
              "<@000000000000000006> *+29.8* **(1185.0)**\n" +
              "<@000000000000000007> *+30.9* **(1067.3)**\n" +
              "<@000000000000000008> *+32.8* **(887.7)**",
            inline: true,
          },
        ],
      },
    ],
    timestamp: "2024-11-26T11:30:00.000000+00:00",
    edited_timestamp: null,
    components: [],
    id: "1310523001611096064",
    channel_id: "1299532381308325949",
    author: {
      id: "857633321064595466",
      username: "NeatQueue",
      avatar: "e803b2f163fda5aeba2cf4820e3a6535",
      discriminator: "0850",
      public_flags: 65536,
      flags: 65536,
      bot: true,
      global_name: null,
    },
    pinned: false,
    mention_everyone: false,
    tts: false,
  },
  timestamp: new Date("2024-11-26T11:30:00.000Z"),
  teams: [
    {
      name: "Eagle",
      players: [
        {
          id: "000000000000000001",
          username: "discord_user_01",
          avatar: "157e517cdbf371a47aaead44675714a3",
          discriminator: "0",
          global_name: "DiscordUser01",
        },
        {
          id: "000000000000000002",
          username: "discord_user_02",
          avatar: null,
          discriminator: "0",
          global_name: "DiscordUser02",
        },
        {
          id: "000000000000000003",
          username: "discord_user_03",
          avatar: null,
          discriminator: "0",
          global_name: null,
        },
        {
          id: "000000000000000004",
          username: "discord_user_04",
          avatar: "6f9ef56a174047263d9c81e9b2559fdc",
          discriminator: "0",
          global_name: "DiscordUser04",
        },
      ],
    },
    {
      name: "Cobra",
      players: [
        {
          id: "000000000000000005",
          username: "discord_user_05",
          avatar: "4081ed914ba463a1374b2b1b11f7bf60",
          discriminator: "0",
          global_name: "DiscordUser05",
        },
        {
          id: "000000000000000006",
          username: "discord_user_06",
          avatar: null,
          discriminator: "0",
          global_name: "DiscordUser06",
        },
        {
          id: "000000000000000007",
          username: "discord_user_07",
          avatar: "d354460036abf8fcddd1282b353bfcd9",
          discriminator: "0",
          global_name: "DiscordUser07",
        },
        {
          id: "000000000000000008",
          username: "discord_user_08",
          avatar: "15c79840f18f08defc64793b55afc2b9",
          discriminator: "0",
          global_name: "DiscordUser08",
        },
      ],
    },
  ],
};

export const channelThreadsResult: RESTPostAPIChannelMessagesThreadsResult = {
  id: "fake-thread-id",
  name: "fake-thread-name",
  type: ChannelType.PublicThread,
  applied_tags: [],
  position: 0,
};
