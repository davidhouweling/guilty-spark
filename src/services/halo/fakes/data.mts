import { MapAsset, MatchStats, PlayerMatchHistory } from "halo-infinite-api";
import { readFile } from "node:fs/promises";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));

async function readMatchStats(filename: string): Promise<[string, MatchStats]> {
  try {
    const fileContents = await readFile(path.join(__dirname, "data", filename), "utf-8");
    const matchStats = JSON.parse(fileContents) as MatchStats;
    return [matchStats.MatchId, matchStats];
  } catch (error) {
    console.error(`Failed to read match stats from ${filename}: ${error as Error}`);
    throw error;
  }
}

async function readPlayerMatches(): Promise<PlayerMatchHistory[]> {
  try {
    const fileContents = await readFile(path.join(__dirname, "data", "player-matches.json"), "utf-8");
    return JSON.parse(fileContents) as PlayerMatchHistory[];
  } catch (error) {
    console.error(`Failed to read player matches: ${error as Error}`);
    throw error;
  }
}

async function readAssetVersion(): Promise<MapAsset> {
  try {
    const fileContents = await readFile(path.join(__dirname, "data", "asset-version.json"), "utf-8");
    return JSON.parse(fileContents) as MapAsset;
  } catch (error) {
    console.error(`Failed to read asset version: ${error as Error}`);
    throw error;
  }
}

export const matchStats = new Map<string, MatchStats>(
  await Promise.all([
    readMatchStats("ctf.json"),
    readMatchStats("koth.json"),
    readMatchStats("land-grab.json"),
    readMatchStats("oddball.json"),
    readMatchStats("slayer.json"),
    readMatchStats("strongholds.json"),
    readMatchStats("total-control.json"),
  ]),
);

export const playerMatches = await readPlayerMatches();

export const assetVersion = await readAssetVersion();

export const playerXuidsToGametags = new Map([
  ["0100000000000000", "gamertag01"],
  ["0200000000000000", "gamertag02"],
  ["0300000000000000", "gamertag03"],
  ["0400000000000000", "gamertag04"],
  ["0500000000000000", "gamertag05"],
  ["0600000000000000", "gamertag06"],
  ["0700000000000000", "gamertag07"],
  ["0800000000000000", "gamertag08"],
  ["0900000000000000", "gamertag09"],
  ["1000000000000000", "gamertag10"],
  ["1100000000000000", "gamertag11"],
  ["1200000000000000", "gamertag12"],
  ["1300000000000000", "gamertag13"],
  ["1400000000000000", "gamertag14"],
  ["1500000000000000", "gamertag15"],
  ["1600000000000000", "gamertag16"],
]);
