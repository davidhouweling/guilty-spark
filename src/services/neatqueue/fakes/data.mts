import { readFile } from "node:fs/promises";
import path from "node:path";
import { fileURLToPath } from "node:url";
import type {
  NeatQueueJoinQueueRequest,
  NeatQueueLeaveQueueRequest,
  NeatQueueMatchCompletedRequest,
  NeatQueueMatchStartedRequest,
  NeatQueueSubstitutionRequest,
  NeatQueueTeamsCreatedRequest,
} from "../types.mjs";

const __dirname = path.dirname(fileURLToPath(import.meta.url));

async function readFakeData<T>(filename: string): Promise<T> {
  const filePath = path.join(__dirname, "./data", filename);
  const fileContents = await readFile(filePath, "utf-8");
  return JSON.parse(fileContents) as T;
}

export const joinQueueData = await readFakeData<NeatQueueJoinQueueRequest>("join-queue.json");
export const leaveQueueData = await readFakeData<NeatQueueLeaveQueueRequest>("leave-queue.json");
export const matchStartedData = await readFakeData<NeatQueueMatchStartedRequest>("match-started.json");
export const teamsCreatedData = await readFakeData<NeatQueueTeamsCreatedRequest>("teams-created.json");
export const substitutionData = await readFakeData<NeatQueueSubstitutionRequest>("substitution.json");
export const matchCompletedData = await readFakeData<NeatQueueMatchCompletedRequest>("match-completed.json");
