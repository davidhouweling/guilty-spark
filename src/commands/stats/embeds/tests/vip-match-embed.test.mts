import { describe, beforeEach, expect, it } from "vitest";
import { matchStats, playerXuidsToGametags } from "../../../../services/halo/fakes/data.mjs";
import { VIPMatchEmbed } from "../vip-match-embed.mjs";
import type { HaloService } from "../../../../services/halo/halo.mjs";
import { Preconditions } from "../../../../base/preconditions.mjs";
import { aFakeHaloServiceWith } from "../../../../services/halo/fakes/halo.fake.mjs";

const match = Preconditions.checkExists(matchStats.get("28af2f64-7c05-458d-b8b1-000000000vip"));

describe("VIPMatchEmbed", () => {
  let haloService: HaloService;
  let matchEmbed: VIPMatchEmbed;

  beforeEach(() => {
    haloService = aFakeHaloServiceWith();
    matchEmbed = new VIPMatchEmbed(haloService);
  });

  describe("getEmbed", () => {
    it("returns the expected embed", async () => {
      const result = await matchEmbed.getEmbed(match, playerXuidsToGametags);
      expect(result).toMatchSnapshot();
    });
  });
});
