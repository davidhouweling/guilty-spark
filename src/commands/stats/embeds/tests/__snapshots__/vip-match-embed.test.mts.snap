// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VIPMatchEmbed > getEmbed > returns the expected embed 1`] = `
{
  "fields": [
    {
      "inline": false,
      "name": "Eagle",
      "value": "Team Score: 9",
    },
    {
      "inline": true,
      "name": "gamertag07",
      "value": "\`\`\`Rank: 5
Score: 5
Kills: 25
Deaths: 11
Assists: 5
KDA: 15.66
Headshot kills: 18
Shots H:F: 270:470
Accuracy: 57.44%
Damage D:T: 6613:4087
Av life duration: 37s
Av damage/life: 601.18
VIP kills: 6
VIP Assists: 1
Kills as VIP: 4
Times selected as VIP: 4
Max killing spree as VIP: 3
Longest Time as VIP: 47s
Time as VIP: 1m 36s\`\`\`",
    },
    {
      "inline": true,
      "name": "gamertag06",
      "value": "\`\`\`Rank: 6
Score: 2
Kills: 14
Deaths: 11
Assists: 6
KDA: 5
Headshot kills: 3
Shots H:F: 251:485
Accuracy: 51.75%
Damage D:T: 6807:4933
Av life duration: 33s
Av damage/life: 618.82
VIP kills: 4
VIP Assists: 3
Kills as VIP: 2
Times selected as VIP: 2
Max killing spree as VIP: 2
Longest Time as VIP: 1m 27s
Time as VIP: 1m 48s\`\`\`",
    },
    {
      "inline": false,
      "name": "
",
      "value": "
",
    },
    {
      "inline": true,
      "name": "gamertag04",
      "value": "\`\`\`Rank: 7
Score: 1
Kills: 3
Deaths: 11
Assists: 8
KDA: -5.33
Headshot kills: 0
Shots H:F: 102:362
Accuracy: 28.17%
Damage D:T: 2721:4748
Av life duration: 33s
Av damage/life: 247.36
VIP kills: 1
VIP Assists: 1
Kills as VIP: 2
Times selected as VIP: 4
Max killing spree as VIP: 1
Longest Time as VIP: 58s
Time as VIP: 2m 39s\`\`\`",
    },
    {
      "inline": true,
      "name": "gamertag05",
      "value": "\`\`\`Rank: 7
Score: 1
Kills: 9
Deaths: 7
Assists: 1
KDA: 2.33
Headshot kills: 3
Shots H:F: 83:207
Accuracy: 40.09%
Damage D:T: 2375:2002
Av life duration: 55s
Av damage/life: 339.29
VIP kills: 1
VIP Assists: 0
Kills as VIP: 0
Times selected as VIP: 0
Max killing spree as VIP: 0
Longest Time as VIP: 
Time as VIP: \`\`\`",
    },
    {
      "inline": false,
      "name": "
",
      "value": "
",
    },
    {
      "inline": false,
      "name": "Cobra",
      "value": "Team Score: 10",
    },
    {
      "inline": true,
      "name": "gamertag03",
      "value": "\`\`\`Rank: 1
Score: 5
Kills: 18
Deaths: 10
Assists: 3
KDA: 9
Headshot kills: 2
Shots H:F: 378:775
Accuracy: 48.77%
Damage D:T: 6247:4444
Av life duration: 37s
Av damage/life: 624.70
VIP kills: 5
VIP Assists: 1
Kills as VIP: 1
Times selected as VIP: 2
Max killing spree as VIP: 1
Longest Time as VIP: 28s
Time as VIP: 45s\`\`\`",
    },
    {
      "inline": true,
      "name": "gamertag02",
      "value": "\`\`\`Rank: 2
Score: 2
Kills: 8
Deaths: 16
Assists: 7
KDA: -5.66
Headshot kills: 3
Shots H:F: 133:352
Accuracy: 37.78%
Damage D:T: 4597:5035
Av life duration: 20s
Av damage/life: 287.31
VIP kills: 2
VIP Assists: 3
Kills as VIP: 3
Times selected as VIP: 5
Max killing spree as VIP: 2
Longest Time as VIP: 1m 14s
Time as VIP: 2m 34s\`\`\`",
    },
    {
      "inline": false,
      "name": "
",
      "value": "
",
    },
    {
      "inline": true,
      "name": "gamertag09",
      "value": "\`\`\`Rank: 2
Score: 2
Kills: 9
Deaths: 7
Assists: 3
KDA: 3
Headshot kills: 7
Shots H:F: 125:246
Accuracy: 50.81%
Damage D:T: 3175:2433
Av life duration: 34s
Av damage/life: 453.57
VIP kills: 2
VIP Assists: 0
Kills as VIP: 2
Times selected as VIP: 1
Max killing spree as VIP: 2
Longest Time as VIP: 1m 6s
Time as VIP: 1m 6s\`\`\`",
    },
    {
      "inline": true,
      "name": "gamertag01",
      "value": "\`\`\`Rank: 4
Score: 1
Kills: 5
Deaths: 14
Assists: 3
KDA: -8
Headshot kills: 0
Shots H:F: 50:214
Accuracy: 23.36%
Damage D:T: 1454:5611
Av life duration: 25s
Av damage/life: 103.86
VIP kills: 1
VIP Assists: 2
Kills as VIP: 3
Times selected as VIP: 5
Max killing spree as VIP: 3
Longest Time as VIP: 1m 13s
Time as VIP: 2m 49s\`\`\`",
    },
    {
      "inline": false,
      "name": "
",
      "value": "
",
    },
    {
      "inline": true,
      "name": "gamertag08",
      "value": "\`\`\`Rank: 9
Score: 0
Kills: 0
Deaths: 4
Assists: 0
KDA: -4
Headshot kills: 0
Shots H:F: 9:55
Accuracy: 16.36%
Damage D:T: 194:1020
Av life duration: 21s
Av damage/life: 48.50
VIP kills: 0
VIP Assists: 0
Kills as VIP: 0
Times selected as VIP: 0
Max killing spree as VIP: 0
Longest Time as VIP: 
Time as VIP: \`\`\`",
    },
    {
      "inline": true,
      "name": "Bot",
      "value": "\`\`\`Rank: 9
Score: 0
Kills: 0
Deaths: 0
Assists: 0
KDA: 0
Headshot kills: 0
Shots H:F: 0:0
Accuracy: 0%
Damage D:T: 0:0
Av life duration: 
Av damage/life: NaN
VIP kills: 0
VIP Assists: 0
Kills as VIP: 0
Times selected as VIP: 0
Max killing spree as VIP: 0
Longest Time as VIP: 
Time as VIP: \`\`\`",
    },
    {
      "inline": false,
      "name": "
",
      "value": "
",
    },
  ],
  "title": "VIP: Fake name for asset 654dff62-d618-496a-8914-06ab73d991e3",
  "url": "https://halodatahive.com/Infinite/Match/28af2f64-7c05-458d-b8b1-000000000vip",
}
`;
